File: 3052 나머지/3052.js
let fs = require('fs');
let inputs = fs.readFileSync('/dev/stdin').toString().split('\n');
inputs.pop()

let answerArray = [];
for (let i = 0; i < 42; ++i) {
    answerArray.push(parseInt(0));
}

for (let i = 0; i < 10; ++i) {
    answerArray[parseInt(inputs[i]) % 42] +=1;
}

let answer = 0;
for (let i = 0; i < 42; ++i) {
    if(answerArray[i] > 0){
        ++answer;
    }
}

console.log(answer);

File: 3052 나머지/3052.py
answerArray = [0 for _ in range(0, 42)]
for i in range(0, 10):
    answerArray[int(input()) % 42] += 1

answer = 0
for i in range(0, 42):
    answer += 1 if answerArray[i] > 0 else 0
print(answer)
